packages/components/src/IconList/src/IconList.tsx

```diff
@@ -0,0 +1,68 @@
+import { Provider, useContextProps } from "react-aria-components";
+import styles from "./IconList.module.css";
+import { IconListContext } from "./IconListContext.ts";
+import { forwardRef, type ForwardedRef, type CSSProperties } from "react";
+import type { BaseComponentProps } from "../../index.ts";
+import clsx from "clsx";
+import { useStyledSystem, type StyledSystemProps, type ResponsiveProp } from "@hopper-ui/styled-system";
+import { IconContext } from "@hopper-ui/icons";
+
+export const GlobalIconListCssSelector = "hop-IconList";
+
+// Won't be needed in next react-aria-components release: https://github.com/adobe/react-spectrum/pull/5850
+const DefaultIconListSlot = "icon";
+
+export interface IconListProps extends StyledSystemProps, BaseComponentProps {
+    /**
+    * The size of the icon.
+    */
+    size?: ResponsiveProp<"sm" | "md" | "lg">;
+}
+
+function IconList(props:IconListProps, ref: ForwardedRef<HTMLSpanElement>) {
+    [props, ref] = useContextProps({ ...props, slot: props.slot || DefaultIconListSlot }, ref, IconListContext);
+
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const { children, style, className, slot, size, ...otherProps } = ownProps;
+
+    const classNames = clsx(
+        className,
+        GlobalIconListCssSelector,
+        styles["hop-iconList"],
```

Comment:
I didn't know we could do something like this... Why are we doing it exactly?

Reply:
We are using CSS Modules, this is why we **have** to use this synthax. I'll write you in slack a couple of things about CSS Modules